### PR TITLE
Patch/cdk ctab

### PR DESCRIPTION
**Pull request by johnmay**
Here's what I had in mind from #325. We move some base/abstract functionality out of cdk-io and into cdk-core/cdk-interfaces. This allows us to split out the MDL formats into a new cdk-ctab module. Also allow us to decouple cdk-smiles from the rest of cdk-io.

Also possible to have a cdk-pubchem, and cdk-sybyl (tripos/mol2 - not sure on name) modules which I'll also add.

Depends on #325

### CODE REVIEW

File: storage/ctab/src/main/java/org/openscience/cdk/io/MDLRXNReader.java [hunk 1]
```diff
@@ -60,11 +60,13 @@
...
  *
  * @cdk.keyword    file format, MDL RXN
+ * @deprecated Use V2000 or V3000
  */
+@Deprecated
 public class MDLRXNReader extends DefaultChemObjectReader {
 
     BufferedReader              input  = null;
-    private static ILoggingTool logger = LoggingToolFactory.createLoggingTool(MDLReader.class);
+    private static ILoggingTool logger = LoggingToolFactory.createLoggingTool(MDLRXNReader.class);
```
Comment (egonw) [inline storage/ctab/src/main/java/org/openscience/cdk/io/MDLRXNReader.java]: *sigh*, that that went unnoticed...

File: storage/ctab/src/main/java/org/openscience/cdk/io/iterator/IteratingSDFReader.java [hunk 2]
```diff
@@ -1,25 +1,24 @@
-/* Copyright (C) 2003-2007  The Chemistry Development Kit (CDK) project
- *                    2014  Mark B Vine (orcid:0000-0002-7794-0426)
+/*
+ * Copyright (c) 2017 John Mayfield <jwmay@users.sf.net>
```
Comment (egonw) [inline storage/ctab/src/main/java/org/openscience/cdk/io/iterator/IteratingSDFReader.java]: Please retain the original copyright statements too.

Comment (johnmay) [inline storage/ctab/src/main/java/org/openscience/cdk/io/iterator/IteratingSDFReader.java]: Doh over optimisitic IDE likes to change these thought I got them all

**Comment by johnmay** [comment]
Rebased and fixed requested issue.

**Comment by egonw** [comment]
Mmm... it seems that with the rebase, it also lost my request?

**Comment by egonw** [comment]
@johnmay it seems that Travis reports a compile issue for the rebased version...

**Comment by johnmay** [comment]
Resolved - your request was https://github.com/cdk/cdk/pull/326/commits/540df0669b6ade16998a76d2f6e25332755b8862

**Pull request merged**